app/src/components/layout/Navbar/Navbar.tsx

```diff
@@ -25,30 +26,39 @@ import { ConfigurableLogo, DefaultNavLogo } from '../LogoComponents';
 
 const navItems = [
   {
-    title: 'home',
+    title: 'Home',
     path: '/',
     key: 'home',
   },
   {
-    title: 'blocks',
+    title: 'Blocks',
     path: '/blocks',
     key: 'blocks',
   },
   {
-    title: 'peers',
+    title: 'Peers',
     path: '/peers',
     key: 'peers',
   },
   {
-    title: 'validators',
+    title: 'Validators',
     path: '/validators',
     key: 'validators',
   },
 ];
 
 export const Navbar: React.FC = () => {
+  const [isOpened, setIsOpened] = useState<boolean>();
+
+  const [selectedRoute, setSelectedRoute] = useState(null);
+
+  const handleNavItemSelection = (event: {
+    target: { innerText: React.SetStateAction<null> };
+  }) => {
+    setSelectedRoute(event.target.innerText);
+  };
```

Comment:
@mgibson-casper @DorianKwan 
Is there a way to make this type work or does this need to be refactored?